discovery_engine_core/core/src/storage.rs

```diff
@@ -52,6 +54,26 @@ pub(crate) trait FeedScope {
     async fn store_documents(&self, documents: &[NewDocument]) -> Result<(), Error>;
 }
 
+#[async_trait]
+pub(crate) trait SearchScope {
+    async fn store_new_search(
+        &self,
+        search: &Search,
+        documents: &[NewDocument],
+    ) -> Result<(), Error>;
+
+    async fn clear(&self) -> Result<bool, Error>;
+
+    async fn fetch(&self) -> Result<(Search, Vec<ApiDocumentView>), Error>;
+
+    async fn store_next_page(
+        &self,
+        page_number: u32,
+        documents: &[NewDocument],
+    ) -> Result<(), Error>;
```

Comment:
🎨 I would move this close to `store_new_search` since they are very similar operations.